openhands/llm/llm.py

```diff
@@ -618,6 +626,8 @@ def format_messages_for_llm(self, messages: Message | list[Message]) -> list[dic
             message.cache_enabled = self.is_caching_prompt_active()
             message.vision_enabled = self.vision_is_active()
             message.function_calling_enabled = self.is_function_calling_active()
+            if self._use_string_serializer:
```

Comment:
```suggestion
            if 'deepseek' in self.config.model:
```

Yes, I'm just suggesting we don't need to save an attribute in llm, we can do this